src/Lazy/Lazy/LazyMultiThreaded.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+
+namespace Lazy
+{
+/// <summary>
+/// Class, implementing multi-threaded lazy calculations for given function
+/// </summary>
+public class LazyMultiThreaded<T> : ILazy<T>
+{
+    private bool _isCalculated;
+    private T? _result;
+    private readonly Func<T> _supplier;
+    private readonly object _lockObject = new();
+
+    public LazyMultiThreaded(Func<T> supplier)
+    {
+        ArgumentNullException.ThrowIfNull(supplier);
+        _supplier = supplier;
+    }
+        
+    public T? Get()
+    {
+        if (_isCalculated) return _result;
```

Comment:
Лучше так:
```suggestion
        if (_isCalculated) 
        {
            return _result;
        }
```
Это длиннее, но делает более очевидной структуру программы